data/typelibrary/utils-1.0.0/typelib/assembling/QUARTERS_TO_LWORD.fct

```diff
@@ -0,0 +1,166 @@
+	IN_14 : BYTE;    // QUARTER BYTE 14
+	IN_15 : BYTE;    // QUARTER BYTE 15
+	IN_16 : BYTE;    // QUARTER BYTE 16
+	IN_17 : BYTE;    // QUARTER BYTE 17
+	IN_18 : BYTE;    // QUARTER BYTE 18
+	IN_19 : BYTE;    // QUARTER BYTE 19
+	IN_20 : BYTE;    // QUARTER BYTE 20
+	IN_21 : BYTE;    // QUARTER BYTE 21
+	IN_22 : BYTE;    // QUARTER BYTE 22
+	IN_23 : BYTE;    // QUARTER BYTE 23
+	IN_24 : BYTE;    // QUARTER BYTE 24
+	IN_25 : BYTE;    // QUARTER BYTE 25
+	IN_26 : BYTE;    // QUARTER BYTE 26
+	IN_27 : BYTE;    // QUARTER BYTE 27
+	IN_28 : BYTE;    // QUARTER BYTE 28
+	IN_29 : BYTE;    // QUARTER BYTE 29
+	IN_30 : BYTE;    // QUARTER BYTE 30
+	IN_31 : BYTE;    // QUARTER BYTE 31
+END_VAR
+
+(* this Function combines the 31 QUARTER BYTES to a lword *)
+QUARTERS_TO_LWORD := LWORD_QUARTER_ZZ;
+
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_31), SHIFT_QUARTER_31);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_30), SHIFT_QUARTER_30);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_29), SHIFT_QUARTER_29);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_28), SHIFT_QUARTER_28);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_27), SHIFT_QUARTER_27);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_26), SHIFT_QUARTER_26);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_25), SHIFT_QUARTER_25);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_24), SHIFT_QUARTER_24);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_23), SHIFT_QUARTER_23);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_22), SHIFT_QUARTER_22);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_21), SHIFT_QUARTER_21);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_20), SHIFT_QUARTER_20);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_19), SHIFT_QUARTER_19);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_18), SHIFT_QUARTER_18);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_17), SHIFT_QUARTER_17);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_16), SHIFT_QUARTER_16);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_15), SHIFT_QUARTER_15);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_14), SHIFT_QUARTER_14);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_13), SHIFT_QUARTER_13);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_12), SHIFT_QUARTER_12);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_11), SHIFT_QUARTER_11);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_10), SHIFT_QUARTER_10);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_09), SHIFT_QUARTER_09);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_08), SHIFT_QUARTER_08);
+QUARTERS_TO_LWORD := QUARTERS_TO_LWORD OR SHL(BYTE_TO_LWORD(IN_07), SHFT_QUARTER_07);
```

Comment:
I assume those are constants?
1.) I think they are missing in the commit, have they be committed in an earlier commit?
2.) the naming convention changes from SHIFT_ to SHFT_ at 07
3.) does it make sense to put those into globally visible constants? I again assume those constants are the values 0 to 31, which makes perfect sense as "magic numbers" in the context of a shift left function

Reply:
Hi, 

the File you name is here: 
https://github.com/Meisterschulen-am-Ostbahnhof-Munchen/4diac_training1/blob/main/Ventilsteuerung/4diacIDE-workspace/test/Type%20Library/utils/conversion-struct/types/const/quarterconst.gcf

we will commit it too. 

i think it is really hard to keep a clear mind if they are not in the global File. 
see how many they are, and how more easy it is to have an overview if they are nicely formatted like this. 

and no, as we talk about 2bits it is constants 0 - 2 - 4 - 6 and so on ... 

about naming we can discuss. SHFT or SHIFT, i guess i wanted the constants File look nice, 
to have the `_` aligned in one Column.